uaclient/apt_news.py

```diff
@@ -90,6 +103,54 @@ def do_selectors_apply(
         if selectors.pro != _is_attached(cfg).is_attached:
             return False
 
+    if selectors.architectures is not None:
+        if system.get_dpkg_arch() not in selectors.architectures:
+            return False
+
+    if selectors.packages is not None:
+        installed_packages = get_installed_packages()
+        installed_packages_names = [
+            package.name for package in installed_packages
+        ]
+        is_package_matched = False
+
+        for package in selectors.packages:
+            try:
+                package_name, version_operator, package_version = package
+            except ValueError:
+                LOG.warning("Invalid package selector: %r", package)
+                return False
+            if not is_installed(package_name):
+                LOG.info("Package %s is not installed", package_name)
+                return False
+            installed_package = installed_packages[
+                installed_packages_names.index(package_name)
+            ]
```

Comment:
instead of looking at the packages list, we can instead use the `get_pkg_version` function directly

Reply:
If we do that, we don't even need the `installed_packages` anymore

Reply:
Thanks @lucasmoura , seems like a good idea.
Updated!